### PR TITLE
💄(ui): Added admin console feature UI files

### PR DESCRIPTION
**Pull request by bevanslabbert**
Generated feature library in libs/client and angular component in libs/client/adminconsole/feature/src/lib.

### CODE REVIEW

File: apps/client/src/app/app.component.html [hunk 1]
```diff
@@ -1 +1,2 @@
-<graduates-shell-feature></graduates-shell-feature>
+<!-- <graduates-shell-feature></graduates-shell-feature> -->
+<graduates-adminconsole></graduates-adminconsole>
```
Comment (KayleePosthumus) [inline apps/client/src/app/app.component.html:2]: Is there a reason this is changed?

**Review by KayleePosthumus — APPROVED** [review]

**Comment by KayleePosthumus** [comment]
@bevanslabbert after you have fixed the conflicts this PR is good to go

**Review by KayleePosthumus — APPROVED** [review]

**Comment by KayleePosthumus** [comment]
@bevanslabbert find out with CI/CD why some of the tests fail and if you can resolve them.

**Comment by Alexander-Muendesi** [comment]
image
@KayleePosthumus @bevanslabbert it seems as though the changes you made to fix those initial conflicts are the reason it is failing

**Comment by Alexander-Muendesi** [comment]
And I think more specifically its to do with changes to the package-lock.json but I am not 100% sure

**Comment by KayleePosthumus** [comment]
@Alexander-Muendesi , @bevanslabbert reverted the file, is there any other reason this might still fail or can we merge?

**Review by KayleePosthumus — APPROVED** [review]

**Comment by Alexander-Muendesi** [comment]
Consulting with CI/CD team on the issue. Will update if any solutions are proposed to your problem

**Comment by arneschreuder** [comment]
Please resolve conflicts.

**Review by KayleePosthumus — DISMISSED** [review]

**Comment by KayleePosthumus** [comment]
@arneschreuder the conflicts are fixed, do you have any idea why it is failing the checks, me and a member of CI/CD are stuck?

**Comment by arneschreuder** [comment]
Lets clear CI/CD cache

**Comment by arneschreuder** [comment]
I suggest closing the PR and then making the PR from a feature branch. Trying to merge fork:develop to upstream:develop can sometimes be tricky.

**Comment by arneschreuder** [comment]
Yeah, the above should def help. @bevanslabbert You need to sync you dev branch with upstream. It is not right.

**Comment by KayleePosthumus** [comment]
@bevanslabbert please follow what arne has said

**Comment by arneschreuder** [comment]
I'm copy pasting this on all the open PRs. Ignore if already seen or applied: https://github.com/COS-301/graduates/discussions/3?sort=new#discussioncomment-2540319

**Comment by arneschreuder** [comment]
Please update with `upstream/develop` and make the PR from a feature branch, not directly from your develop. It makes it difficult for you if issues occur.

File: apps/client/src/app/app.module.ts [hunk 2]
```diff
@@ -6,6 +6,8 @@ import { ServiceWorkerModule } from '@angular/service-worker';
 import { FeatureModule as ClientShellFeatureModule } from '@graduates/client/shell/feature';
 import { environment } from '../environments/environment';
 import { AppComponent } from './app.component';
+import { ClientAdminconsoleFeatureModule as AdminConsole } from '@graduates/client/adminconsole/feature';
```
Comment (arneschreuder) [inline apps/client/src/app/app.module.ts:9]: No, this is wrong. Your feature should be lazy loading using angular routing. https://angular.io/guide/lazy-loading-ngmodules

File: libs/client/adminconsole/feature/src/lib/adminconsole/adminconsole.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { AdminconsoleComponent } from './adminconsole.component';
+
+describe('AdminconsoleComponent', () => {
```
Comment (arneschreuder) [inline libs/client/adminconsole/feature/src/lib/adminconsole/adminconsole.component.spec.ts:5]: Needs a LOT more testing.

File: libs/client/adminconsole/feature/src/lib/adminconsole/adminconsole.component.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+@Component({
+  selector: 'graduates-adminconsole',
+  templateUrl: './adminconsole.component.html',
+  styleUrls: ['./adminconsole.component.scss']
+})
+export class AdminconsoleComponent {
+
+
+  burger : boolean
```
Comment (arneschreuder) [inline libs/client/adminconsole/feature/src/lib/adminconsole/adminconsole.component.ts:11]: This is fine for now. But please make use of a store to do state and data management for your feature. https://www.ngxs.io.

File: libs/client/adminconsole/feature/src/lib/client-adminconsole-feature.module.spec.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+import { ClientAdminconsoleFeatureModule } from './client-adminconsole-feature.module';
+
+describe('ClientAdminconsoleFeatureModule', () => {
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ClientAdminconsoleFeatureModule],
+    }).compileComponents();
+  }));
+
+  // TODO: Add real tests here.
```
Comment (arneschreuder) [inline libs/client/adminconsole/feature/src/lib/client-adminconsole-feature.module.spec.ts:11]: Yes, I agree :)

File: libs/client/adminconsole/feature/src/lib/client-adminconsole-feature.module.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { AdminconsoleComponent } from './adminconsole/adminconsole.component';
+
+@NgModule({
+  imports: [CommonModule],
```
Comment (arneschreuder) [inline libs/client/adminconsole/feature/src/lib/client-adminconsole-feature.module.ts:6]: So this module must import from a `client-adminconsole-feature.routing-module.ts` which does the lazy loading for this feature.

File: package.json [hunk 7]
```diff
@@ -99,4 +99,4 @@
     "ts-jest": "27.0.5",
     "typescript": "~4.5.2"
   }
-}
+}
```
Comment (arneschreuder) [inline package.json:102]: Please chuck this away in your PR.

File: tsconfig.base.json [hunk 8]
```diff
@@ -233,4 +236,4 @@
     }
   },
   "exclude": ["node_modules", "tmp"]
-}
+}
```
Comment (arneschreuder) [inline tsconfig.base.json:239]: Please add new line.

**Review by arneschreuder — CHANGES_REQUESTED** [review]
Please do not use npm. Use yarn. Remove package-lock.json. This should also be gitignored.

**Comment by arneschreuder** [comment]
PR also still has conflicts. Remember to sync with upstream/develop.

**Comment by bevanslabbert** [comment]
Thank you for the feedback!

**Pull request closed without merging**